### PR TITLE
Implement per-function profiling metrics for asp

### PR DESCRIPTION
**Pull request by peterebden**
Gives some little subdivisions on hot functions about how long we've spent in each one because of a particular build function.

It's a little limited since we only spend just so long in each one, but still seems nice to have (and it's not very hard to do)

### CODE REVIEW

File: src/please.go [hunk 1]
```diff
@@ -991,6 +991,7 @@ func readConfigAndSetRoot(forceUpdate bool) *core.Configuration {
 	}
 	config := readConfig(forceUpdate)
 	// Now apply any flags that override this
+	config.Profiling = opts.Profile != ""
```
Comment (Tatskaari) [inline src/please.go:994]: Perhaps we should set the profile file to some default value (`please.prof`?) if `config.Profile` has been set.

**Review by Tatskaari — APPROVED** [review]
Nice! Does this hook into the same dot file as the normal please profile or does it produce it's own call graph? I'm concerned it might double count stuff if it's profiling the interpreter interpreting the asp code AND the asp code itself.

File: src/please.go (hunk 1 above)
Comment (Tatskaari) [reply src/please.go:994]: Ah, I see, this is a psudo config value that can't be set in the `.plzconfig`. Think this makes sense.

**Comment by peterebden** [comment]
Yeah it appears in the same profile file - it doesn't count twice, it's more like it subdivides the existing CPU time into these other slices. I think it is just strictly adding information.

**Comment by Tatskaari** [comment]
> Yeah it appears in the same profile file - it doesn't count twice, it's more like it subdivides the existing CPU time into these other slices. I think it is just strictly adding information.

Well that sounds perfect! Nice one :)

**Review by tiagovtristao — APPROVED** [review]
Nice!

**Pull request merged**